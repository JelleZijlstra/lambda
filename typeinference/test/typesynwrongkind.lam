data either a b = Left a | Right b in
type with_err_code = either int in

let hd : with_err_code = Right 3 in
()
