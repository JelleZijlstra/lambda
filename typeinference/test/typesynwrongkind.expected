%%stdout%%
Error: Cannot unify types: either/0 and either/0 'typevar/0
%%stderr%%
